packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -325,7 +331,13 @@ export function set_dynamic_element_attributes(node, prev, next, css_hash) {
 		}
 
 		for (key in next) {
-			set_custom_element_data(node, key, next[key]);
+			// TODO do we need to separate attributes from properties here? probably
```

Comment:
On reflection, I think the existing heuristic is probably fine in this context — if you don't know what the element is at author time, then you can't know which things are props and which aren't _anyway_. In any case it's the edgiest of edge cases

```suggestion
```